internal/prober/interpolation/interpolation.go

```diff
@@ -0,0 +1,235 @@
+package interpolation
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/grafana/synthetic-monitoring-agent/internal/model"
+	"github.com/rs/zerolog"
+)
+
+// VariableRegex matches ${variable_name} patterns
+var VariableRegex = regexp.MustCompile(`\$\{([a-zA-Z_][a-zA-Z0-9_]*)\}`)
+
+// SecretRegex matches ${secrets.secret_name} patterns
+var SecretRegex = regexp.MustCompile(`\$\{secrets\.([^}]*)\}`)
+
+// VariableProvider defines the interface for resolving variables
+type VariableProvider interface {
+	GetVariable(name string) (string, error)
+}
+
+// SecretProvider defines the interface for resolving secrets
+type SecretProvider interface {
+	GetSecretValue(ctx context.Context, tenantID model.GlobalID, secretKey string) (string, error)
+}
+
+// Resolver handles string interpolation for both variables and secrets
+type Resolver struct {
+	variableProvider VariableProvider
+	secretProvider   SecretProvider
+	tenantID         model.GlobalID
+	logger           zerolog.Logger
+	secretEnabled    bool
+}
+
+// NewResolver creates a new interpolation resolver
+func NewResolver(variableProvider VariableProvider, secretProvider SecretProvider, tenantID model.GlobalID, logger zerolog.Logger, secretEnabled bool) *Resolver {
+	return &Resolver{
+		variableProvider: variableProvider,
+		secretProvider:   secretProvider,
+		tenantID:         tenantID,
+		logger:           logger,
+		secretEnabled:    secretEnabled,
+	}
+}
+
+// Resolve performs string interpolation, replacing both variables and secrets
+func (r *Resolver) Resolve(ctx context.Context, value string) (string, error) {
+	if value == "" {
+		return "", nil
+	}
+
+	// First resolve secrets if enabled
+	if r.secretEnabled {
+		resolvedValue, err := r.resolveSecrets(ctx, value)
+		if err != nil {
+			return "", err
+		}
+		value = resolvedValue
+	}
+
+	// Then resolve variables
+	if r.variableProvider != nil {
+		resolvedValue, err := r.resolveVariables(value)
+		if err != nil {
+			return "", err
+		}
+		value = resolvedValue
+	}
```

Comment:
I'm not following.

Why does resolving the secret and then using *that* value to resolve variables make sense?